### PR TITLE
Update programs test

### PR DESCRIPTION
**Pull request by pattyshack**
- dedup test setup
- switch to vm.RunV2
- switch to storage.SnapshotTree

### CODE REVIEW

File: fvm/environment/programs_test.go [hunk 1]
```diff
@@ -122,58 +138,60 @@ func Test_Programs(t *testing.T) {
...
+			contractALocation)
 		require.NoError(t, err)
 
 		require.Equal(t, contractACode, string(retrievedContractA))
 
 	})
-
 	t.Run("register touches are captured for simple contract A", func(t *testing.T) {
-
-		// deploy contract A
```
Comment (pattyshack) [inline fvm/environment/programs_test.go:161]: this was a bug in the test.  we've already deploy contract A (immediately above), and didn't check for output.Error here.

File: fvm/environment/programs_test.go [hunk 2]
```diff
@@ -122,58 +138,60 @@ func Test_Programs(t *testing.T) {
...
+			contractALocation)
 		require.NoError(t, err)
 		require.Empty(t, retrievedContractA)
 
 		// deploy contract A0
-		procContractA0 := fvm.Transaction(
-			contractDeployTx("A", contractA0Code, addressA),
-			derivedBlockData.NextTxIndexForTestingOnly())
-		err = vm.Run(context, procContractA0, mainView)
+		executionSnapshot, output, err := vm.RunV2(
```
Comment (SaveTheRbtz) [inline fvm/environment/programs_test.go:151]: linter has a point here about output not being used.

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/environment/programs_test.go (hunk 2 above)
Comment (pattyshack) [reply fvm/environment/programs_test.go:151]: thx.  i deleted the line that checked the output.  it should be an output.Err should not be nil

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**